### PR TITLE
update conditionals notebook

### PR DESCRIPTION
**Pull request by bradleyboehmke**
- Made cosmetic changes to how pseudo code examples were being displayed
- Changed Your Turn example that is built upon in each section (if; if-else; if-elif-else)
- Added a short section that shows different variants of common conditional statement variants
    - conditioned on type of object (`isinstance`)
    - conditioned on if something is in (`in`) or not in (`not in`)
    - conditioned on if something is (empty vs non-empty objects and the idea of 'truthiness')
    - conditioned on if some or all values are true (`any`, `all`)
- Added a slide that illustrates the idea of vectorization

### CODE REVIEW

**Comment by eswan18** [comment]
This looks great! Really good additions in the conditional variants and the vectorization. A couple small things:

- The first slide titled *Vectorization* doesn't fit on the screen – the second image is almost entirely cut off on my 15" MacBook Pro.
- Some of the Python-code-in-markdown renders with weird indentation when I convert this notebook to slides. I'm guessing it doesn't happen in slide mode for you, so as long as you present this section from your computer, it shouldn't be a problem.

So I think you should take a look at the image sizing in the first issue before we merge this (maybe it doesn't happen on your side, using Rise?), but otherwise it's good to go. Thanks!

**Comment by bradleyboehmke** [comment]
Good catch on the image size, I was viewing it on my large monitor so didn't even think about the sizing concern.  Also, the markdown python code renders as below on my end (when converted to slide format):

Screen Shot 2021-02-18 at 6 13 53 AM

In the document I actually start the markdown block 3 spaces from the left so that it aligns with the bullet that precedes it. Within the markdown block the indentation should be 4 spaces. I did notice a few straggling cases where the indentation was 5 spots rather than 4 and made those updates. If it continues to be a problem I can see if its being caused by the fact that I'm indenting the markdown block to align to the bullets.

**Comment by eswan18** [comment]
Here's an example of the weird indentation, which is probably only happening in `jupyter nbconvert`
image

And for me, the size change doesn't seem to have had any effect. Again, `jupyter nbconvert` seems not to display things in the same way they're showing up on your side – but that's fine, if you present this section it won't matter.
image

Feel free to merge this.

**Review by eswan18 — APPROVED** [review]

**Pull request merged**
